src/supervisor/watchers/handlers/index.ts

```diff
@@ -109,6 +110,20 @@ export function setupInformer(namespace: string, workloadKind: WorkloadKind) {
 
   const informer = makeInformer<KubernetesObject>(kubeConfig, namespacedEndpoint, loggedListMethod);
 
+  informer.on(ERROR, (err) => {
+    // Types from client library insists that callback is of type KubernetesObject
+    if ((err as any).code === ECONNRESET_ERROR_CODE) {
```

Comment:
I think I'd be fine comparing directly to `'ECONNRESET'`, up to you! We actually compare to these network codes in another place too (`retryKubernetesRequest()` or whatever the name of the function).

Reply:
I think this is too specific a place and don't involve retries like the retry function as well as that function has very specific cases to querying k8s API